### ISSUE TITLE
"View source" freezes the client

### ISSUE THREAD

**Issue by yetanotherus3r** [ValveSoftware/steam-for-linux#3616]
Right click -> View source in any of the Store, Community, or Profile pages causes the client to hang up loading the page until the client is closed and restarted. The client remains responsive for a while (you can access Library, but not any other tab and the traybar icon) until it hard freezes, requiring a kill -9.

The offending steamwebhelper process takes 100% CPU and ridicolous amount of memory (>3 GB). Killing it also usually un-freezes the client if done fast.
The command line options showing up in System Monitor are
steamwebhelper -cefhost -cachedir config/htmlcache -cookiedir config/cookies -steampid 9307 --blacklist-accelerated-compositing --process-per-tab --disable-accelerated-video-decode

Happens on Ubuntu 14.10 with Steam package 1417546230

:confused:

**Comment by NoXPhasma** [comment]
I can confirm this too. Steam gets unresponsive, one CPU core gets to 100% and the memory gets filled. So there is a memory leak too.

**Comment by alfred-valve** [comment]
I've found the issue and we will work on a fix

**Comment by MrSchism** [comment]
This is no longer unresponsive, but rather it spits an interesting error (at least on my system):

`ATTENTION: default value of option force_s3tc_enable overridden by environment.`

It opens the application normally used, but the application itself doesn't always get the source.  Perhaps opening a new Steam window might work best.

**Comment by alfred-valve** [comment]
The "view source" open does a "xdg-open /tmp/<file name>,txt" . The original bug was that it didn't run that command detached, so you would then freeze the main steam loop, so we fixed it to run in the background. I'm not sure why your app isn't opening the file, use "ps aux" to find the xdg-open command line and see if you can work out what part of it is failing?

**Comment by MrSchism** [comment]
I'm fairly sure the application I'm using ([Brackets](http://brackets.io/)) is at fault.  I'm opening an issue with them because of this.

For clarity to what I noticed and why this may eventually need another workaround:
Instead of populating a blank file, it does't do anything after opening.  When a second attempt occurs with Brackets still open from the first time, it simply opens another instance of Brackets.  This points me to it being an error on their end.

That being said, it's not the first application I've used to display this behavior in similar circumstances.

I'm going to attempt to replicate it with other editors.
